pkg/cli/cli.go

```diff
@@ -151,21 +151,26 @@ func (c *CLI) buildCmd() error {
 	switch err := c.getInfo(); {
 	case err == nil:
 	case errors.As(err, &uve) && uve.Version.Compare(config.Version{Number: 3, Stage: stage.Alpha}) == 0:
-		// Check if the corresponding stable version exists, set c.projectVersion and break
 		stableVersion := config.Version{
 			Number: uve.Version.Number,
 		}
 		if config.IsRegistered(stableVersion) {
-			// Use the stableVersion
 			c.projectVersion = stableVersion
 		} else {
-			// stable version not registered, let's bail out
 			return err
 		}
 	default:
 		return err
 	}
 
+	// Workaround for kubebuilder alpha generate
+	if c.cmd.CalledAs() == "alpha generate" {
+		err := updateProjectFileForAlphaGenerate()
+		if err != nil {
+			return fmt.Errorf("failed to update PROJECT file: %w", err)
```

Comment:
We have tests here: https://github.com/kubernetes-sigs/kubebuilder/tree/master/test/e2e/alphagenerate

We can create a new project and add 1 API
Then, update the PROJECT file to replace `go.kubebuilder.io/v4` with go.kubebuilder.io/v2 or go.kubebuilder.io/v3 and then call the command with the flag that --force OR allow upgrade the PROJECT file without the need to say Y so we can ensure that works fine. 

WDYT?